src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1481,7 +1481,7 @@ private BoundExpression BindDefaultExpression(DefaultExpressionSyntax node, Diag
                 expression = null;
                 if (node is IdentifierNameSyntax identifier)
                 {
-                    var type = BindNativeIntegerSymbolIfAny(identifier, diagnostics);
+                    var type = IsInsideNameof ? null : BindNativeIntegerSymbolIfAny(identifier, diagnostics);
```

Comment:
`int` isn't allowed in nameof. However, something like `nameof(int.MaxValue)` is allowed.

Is there similar thing for nint? e.g. `nameof(nint.Something)` which should be allowed?

If there is something like that, my approach will probably fail.